homeassistant/components/tplink/light.py

```diff
@@ -61,19 +63,34 @@
 async def async_setup_entry(hass: HomeAssistantType, config_entry, async_add_entities):
     """Set up lights."""
     devices = hass.data[TPLINK_DOMAIN][CONF_LIGHT]
-    entities = []
-
-    await hass.async_add_executor_job(get_devices_sysinfo, devices)
-    for device in devices:
-        entities.append(TPLinkSmartBulb(device))
-
-    async_add_entities(entities, update_before_add=True)
+    entities_ready = []
+    entities_unavailable = []
+
+    entities_ready, entities_unavailable = await hass.async_add_executor_job(
```

Comment:
Except for the entity class, this code is the same as `switch.py`. Please reuse.